.github/actions/run-ios-simulator/src/main.ts

```
@@ -0,0 +1,74 @@
+        const matches = runtimeId.match(/(iOS \d{1,2}(.\d{1,2})?)/g);
+        if (matches && matches.length > 0) {
+            runtimeId = matches[0].replace(" ", "");
+            core.info(`runtimeId: ${runtimeId}`);
+        }
+
+        try {
+            await execCmd(`xcrun simctl create ${id} com.apple.CoreSimulator.SimDeviceType.${iphoneToSimulate} ${runtimeId.toString()}`);
+        }
+        catch {
+            // Different combinantions of xcode and macOS versions have shown different syntax acceptance about the runtime, therefore 1 last attempt with a different syntax.
+            // additionally, this uses  childProcess.exec instead of @actions/exec because the latter doesn't properly parse the pipe (|) char
+            const { stdout, stderr } = await childProcess.exec("xcrun simctl list runtimes |  awk '/com.apple.CoreSimulator.SimRuntime.iOS/ { match($0, /com.apple.CoreSimulator.SimRuntime.iOS-[0-9.-]+/); print substr($0, RSTART, RLENGTH); exit }'");
+            runtimeId = stdout?.toString() ?? "";
+            if (stderr) {
+                core.setFailed(stderr.toString());
+            }
+
+            await execCmd(`xcrun simctl create ${id} com.apple.CoreSimulator.SimDeviceType.${iphoneToSimulate} ${runtimeId.toString()}`);
+        }
+
+        await execCmd(`xcrun simctl boot ${id}`);
+        await execCmd(`xcrun simctl install ${id} ${appPath}`);
+        await execCmd(`xcrun simctl launch --console-pty ${id} ${bundleId} ${args}`);
+    } catch (error) {
+        core.setFailed(error.message);
+    } finally {
+        try {
+            await execCmd(`xcrun simctl shutdown ${id}`);
+            await execCmd(`xcrun simctl delete ${id}`);
+        } catch (error) {
+            core.setFailed(`An error occurred during cleanup: ${error.toString()}`);
+        }
+    }
+}
+
+async function execCmd(cmd: string, options: exec.ExecOptions = {}): Promise<void> {
```

Can we modify this to return the output from `stdout`?

---

So would you like to see a tuple like `result: [number, string]` that returns the exit code and the stdout, or do you suggest just the _stdout_?

---

I think it should internally capture both stdout and stderr and return just stdout. If the exit code is non-zero, it should throw an exception that includes the output of stderr.